### PR TITLE
feat(octavia): health monitor creation

### PR DESCRIPTION
**Pull request by qpavy**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/octavia-health-monitor`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-10689
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

## Description

Add health monitor creation form inside octavia load balancer pool

### CODE REVIEW

**Review by antonymarion — APPROVED** [review]

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/constants.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+export const LABELS = {
+  MAX_RETRIES_DOWN: 'Max Retries Down',
+  MAX_RETRIES: 'Max Retries',
+  PERIODICITY: 'Periodicity',
+  TIMEOUT: 'Timeout',
+  URLPATH: 'URL Path',
+};
+
+export const MINMAX = {
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/constants.js:9]:
```suggestion
export const BOUNDS = {
```
or whatever you want but `MINMAX` is hard to read

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "octavia_load_balancer_health_monitor_form_name": "Nom",
+  "octavia_load_balancer_health_monitor_form_name_help": "Doit uniquement contenir des nombres, lettres, underscores, tirets ou points.",
+  "octavia_load_balancer_health_monitor_form_type": "Type",
+  "octavia_load_balancer_health_monitor_form_type_default": "Choisissez un type",
+  "octavia_load_balancer_health_monitor_form_expected_code": "Code attendu",
+  "octavia_load_balancer_health_monitor_form_expected_code_help": "Le code HTTP considéré comme un succès",
+  "octavia_load_balancer_health_monitor_form_max_retires_down_tooltip": "Le nombre d'échecs de connexion autorisés avant de marquer le membre comme erreur. Doit être un nombre compris entre 1 et 10. La valeur par défaut est 3. ",
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/translations/Messages_fr_FR.json:8]:
```suggestion
  "octavia_load_balancer_health_monitor_form_max_retires_down_tooltip": "Le nombre d'échecs de connexion autorisé avant de marquer le membre comme erreur. Doit être un nombre compris entre 1 et 10. La valeur par défaut est 3. ",
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/create/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -1,6 +1,8 @@
 {
-  "octavia_load_balancer_health_monitor_title": "Ajouter un health monitor",
-  "octavia_load_balancer_health_monitor_description_1": "Le health monitor est utilisé pour déterminer l’intégrité des membres de votre pool. Des contrôles d’intégrité sont exécutés régulièrement sur chaque membre du pool et le résultat de ces contrôles est utilisé pour déterminer si le membre reçoit de nouvelles connexions. Un seul health monitor peut être défini pour chaque pool.",
-  "octavia_load_balancer_health_monitor_description_2": "Découvrez des options additionnelles via Horizon.",
-  "octavia_load_balancer_health_monitor_description_link": "Découvrir des options"
+  "octavia_load_balancer_health_monitor_create_title": "Ajouter un health monitor",
+  "octavia_load_balancer_health_monitor_create_description_1": "Le health monitor est utilisé pour déterminer l’intégrité des membres de votre pool. Des contrôles d’intégrité sont exécutés régulièrement sur chaque membre du pool et le résultat de ces contrôles est utilisé pour déterminer si le membre reçoit de nouvelles connexions. Un seul health monitor peut être défini pour chaque pool.",
```
Comment (dectotam) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/create/translations/Messages_fr_FR.json:3]:
```suggestion
  "octavia_load_balancer_health_monitor_create_description_1": "Le health monitor est utilisé pour déterminer l'intégrité des membres de votre pool. Des contrôles d'intégrité sont exécutés régulièrement sur chaque membre du pool et le résultat de ces contrôles est utilisé pour déterminer si le membre reçoit de nouvelles connexions. Un seul health monitor peut être défini pour chaque pool.",
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/constants.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  PERIODICITY: {
+    MIN: 1,
+  },
+  TIMEOUT: {
+    MIN: 0,
+  },
+};
+
+export const HEALTH_MONITOR_NAME_REGEX = /^[A-Za-z0-9_.-]+$/;
+export const EXPECTED_STATUS_CODE_REGEX = /^[1-5][0-9][0-9]$/;
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/constants.js:27]: Could be simplified
```suggestion
export const EXPECTED_STATUS_CODE_REGEX = /^[1-5][0-9]{2}$/;
```

**Review by dectotam — CHANGES_REQUESTED** [review]

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/controller.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      this.model.urlPath = '/';
+      this.model.expectedCode = 200;
+    } else {
+      this.displayHttpSpecificFields = false;
+      delete this.model.urlPath;
+      delete this.model.expectedCode;
+    }
+  }
+
+  static isTypeIsHttpOrHttps(type) {
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/controller.js:53]: I would have named this function isTypeHttpOrHttps

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/create/translations/Messages_fr_FR.json (hunk 3 above)
Comment (antonymarion) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/create/translations/Messages_fr_FR.json:3]: As I always put ` in my french, I could not see that one :)

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/service.js [hunk 6]
```diff
@@ -4,12 +4,41 @@ export default class OctaviaLoadBalancerHealthMonitorService {
...
+        urlPath: healthMonitor.urlPath,
+      };
+    }
+
+    return this.$http
+      .post(
+        `/cloud/project/${projectId}/region/${region}/loadbalancing/healthMonitor`,
+        postDatas,
+      )
+      .then(() => this.broadcast('useraccount.security.users.refresh', {}));
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/service.js:42]: Where is defined `broadcast` function ?

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  "octavia_load_balancer_health_monitor_form_periodicity": "Intervalle",
+  "octavia_load_balancer_health_monitor_form_periodicity_tooltip": "Intervalle entre 2 tests du Health Monitor. Doit être supérieur ou égal au timeout.",
+  "octavia_load_balancer_health_monitor_form_max_retries_tooltip": "Le nombre d'échecs de connexion autorisés avant de marquer le membre comme inactif. Doit être un nombre compris entre 1 et 10.",
+  "octavia_load_balancer_health_monitor_form_timeout_tooltip": "Le temps après lequel un health check s'arrête. Doit être un nombre supérieur ou égal à 0 et inférieur à la périodicité",
+  "octavia_load_balancer_health_monitor_form_cancel": "Annuler",
+  "octavia_load_balancer_health_monitor_form_submit": "Editer"
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/translations/Messages_fr_FR.json:14]: Shouldn't it be "Ajouter" ?

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/create/component.js [hunk 8]
```diff
@@ -2,6 +2,13 @@ import template from './template.html';
 import controller from './controller';
 
 export default {
+  bindings: {
+    projectId: '<',
+    region: '<',
+    poolId: '<',
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/create/component.js:8]: poolId is a bit redundant, given you have the pool

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/service.js (hunk 6 above)
Comment (qpavy) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/service.js:42]: wtf I don't know where this come from I remove it

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/constants.js (hunk 1 above)
Comment (qpavy) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/constants.js:9]: it comes from css minmax properties :) I replace it

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/create/component.js (hunk 8 above)
Comment (qpavy) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/create/component.js:8]: I get it from pool routing resolve where it is already redundant :)

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/translations/Messages_fr_FR.json (hunk 7 above)
Comment (qpavy) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/health-monitor/components/health-monitor-form/translations/Messages_fr_FR.json:14]: Nop I change the way the common components work by giving him a submit label. If no submit label is given, the default label is Editer. But is that case , a submit label is send by parent

**Review by dectotam — APPROVED** [review]

**Review by JacquesLarique — APPROVED** [review]

**Pull request merged**